pandas/tests/series/test_asof.py

```diff
@@ -148,3 +150,8 @@ def test_errors(self):
         s = Series(np.random.randn(N), index=rng)
         with self.assertRaises(ValueError):
             s.asof(s.index[0], subset='foo')
+
+        # series is all nans
+        result = Series([np.nan]).asof([0])
```

Comment:
Can you make this a separate test? (as it is not related to errors). Eg `test_all_nans`